load-balancer/response-time/src/main/java/io/smallrye/stork/loadbalancer/leastresponsetime/impl/CallStatistics.java

```diff
@@ -0,0 +1,83 @@
+package io.smallrye.stork.loadbalancer.leastresponsetime.impl;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.concurrent.atomic.AtomicLong;
+
+import io.smallrye.stork.loadbalancer.leastresponsetime.impl.util.FastPower;
+import io.smallrye.stork.spi.CallStatisticsCollector;
+
+public class CallStatistics implements CallStatisticsCollector {
+
+    private final AtomicLong callCount = new AtomicLong(1);
+    private final ConcurrentHashMap<Long, CallsData> storage = new ConcurrentHashMap<>();
+
+    final long errorPenalty;
+    final FastPower powersOfDecliningFactor;
+
+    public CallStatistics(long errorPenalty, FastPower powersOfDecliningFactor) {
+        this.errorPenalty = errorPenalty;
+        this.powersOfDecliningFactor = powersOfDecliningFactor;
+    }
+
+    @Override
+    public void storeResult(long id, long timeInNs, Throwable error) {
```

Comment:
It doesn't store/record the score but a result of a single call made by the client library. I think `recordScore` could be confusing. E.g. the power of 2's doesn't have a score, it just keeps track of inflight requests

Reply:
Ok, maybe some javadoc can describe it then.

Reply:
> 

Ok, maybe some javadoc can describe it then.

Reply:
> 
It doesn't store/record the score but a result of a single call made by the client library. I think `recordScore` could be confusing. E.g. the power of 2's doesn't have a score, it just keeps track of inflight requests
>

Ok, in that case we could add some javadoc to describe it, wdyt?

Reply:
there is [some](https://github.com/smallrye/smallrye-stork/blob/main/api/src/main/java/io/smallrye/stork/spi/CallStatisticsCollector.java#L15) but we can  extend it

Reply:
I renamed the artifact and the directory. Left the md file as it was, we can rename it later. I also added a CHANGELOG.md file because I didn't know where better to store the info of a breaking change